pyproject.toml

```diff
@@ -95,6 +95,7 @@ pyspark = "3.5.3"
 cython = "3.0.11"
 deptry = ">=0.14,<0.22"
 docutils = "!=0.21.post1"   # https://github.com/python-poetry/poetry/issues/9248#issuecomment-2026240520
+ruff = "^0.8.6"
```

Comment:
In the past we kept the versions of `pre-commit` and the `pyproject.toml` separate, could you elaborate on why we should add this here?

Reply:
wanted to be able to run ruff locally, `poetry run ruff ...` but i guess `make lint` will suffice

Reply:
I see, but this might ignore configuration that we pass in the `pre-commit` yaml